runtime/compiler/CMakeLists.txt

```diff
@@ -291,8 +290,16 @@ else()
 	message(SEND_ERROR "unsupported platform")
 endif()
 
-omr_stringify(J9_CFLAGS_STR ${J9_CFLAGS} ${J9_SHAREDFLAGS})
-omr_stringify(J9_CXXFLAGS_STR ${J9_CXXFLAGS} ${J9_SHAREDFLAGS})
+if(J9VM_OPT_JITSERVER)
+	list(APPEND J9_CXXFLAGS -std=c++11)
+endif()
+
+# Add optional user-specified compiler flags that only apply to the JIT
+list(APPEND J9_CFLAGS ${J9JIT_EXTRA_CFLAGS})
+list(APPEND J9_CXXFLAGS ${J9JIT_EXTRA_CXXFLAGS})
+
+omr_stringify(J9_CFLAGS_STR ${J9_SHAREDFLAGS} ${J9_CFLAGS})
+omr_stringify(J9_CXXFLAGS_STR ${J9_SHAREDFLAGS} ${J9_CXXFLAGS})
```

Comment:
Because otherwise the extra `-O` and `-g` flags added to `OMR_PLATFORM_CXX_COMPILE_OPTIONS` are overridden by the default ones in `OMR_PLATFORM_COMPILE_OPTIONS`. `J9_SHAREDFLAGS` is derived from `OMR_PLATFORM_COMPILE_OPTIONS`, and `J9_CXXFLAGS` is derived from `OMR_PLATFORM_CXX_COMPILE_OPTIONS`. The new order will also match the rest of the VM build that appends these flags in the order established here: https://github.com/eclipse/omr/blob/master/cmake/modules/OmrPlatform.cmake#L54-L64.

Reply:
Consistency is nice, but reordering conflicting options may yield unexpected results.
I'd like @mpirvu (or his delegate) to weigh in on this.

Reply:
@ymanton Since you have more knowledge than me on the building mechanism, could you please offer your opinion on the topic above? Thanks

Reply:
If there's an implicit ordering dependency between the options in `J9_CFLAGS`/`J9_CXXFLAGS` and the options in `J9_SHAREDFLAGS` (or the OMR counterparts) I'd consider that more of a problem than the change that exposed it by simply re-ordering them.

Reply:
I took a quick look at the kinds of options that we're putting in those lists and I see mostly independent flags and options so we're probably OK.

Reply:
How do we improve on 'probably'? What can we do to be sure the reordering is not problematic?

Reply:
I did verify that at least for GCC no conflicting options are passed during the build. Looking at what options are added to `J9_*FLAGS` for other platforms in `compiler/CMakeLists.txt`, almost all of them are `-D` defines, and all appear to be independent. If there were any conflicting options in `OMR_PLATFORM_*COMPILE_OPTIONS`, the JIT would get a different version from the rest of the VM, and this PR fixes this inconsistency. I agree with @ymanton that having conflicting options would be a bigger problem on its own.